chef/automate
# Refactored mock server service code

### :nut_and_bolt: Description: What code changed, and why?
Refactored Start and Stop code for mock server
Added  start and stop HTTP server


### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-3442

### :+1: Definition of Done

### :athletic_shoe: How to Build and Test the Change

- Take the pull of this PR
- rebuild components/automate-cli/
- Install that cli build on the ubuntu or any linux supported by Automate
- Run chef-automate verify serve
- Curl request for start :- `curl --location 'http://<machine-ip>:7799/api/v1/start/mock-server' \
--header 'Content-Type: application/json' \
--data '{
  "port": <port>,
  "protocol": <protocol>,
  "cert":<cert-for-https>,
  "key":<key-for-https>
}'`
-Curl request for stop :- `curl --location 'http://<machine-ip>:7799/api/v1/stop/mock-server' \
--header 'Content-Type: application/json' \
--data '{
  "port": <port>,
  "protocol": <protocol>,
}'`

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
[start-stop-mockserver-using-curl-concurrently](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/Ebj1Y9jJ8GVOsDHjez2fAlABJWdSPxkSClsumDt7LyJQqg?e=9E6Xau)
[start-stop-mockserver-api](https://progresssoftware.sharepoint.com/:v:/s/ChefCoreC/EU3xr1iptsdNghnab_u8cBEBGk4NDOIj6v9x5ZUUH1K-CQ?e=vbnTY1)

<img width="1172" alt="Screenshot 2023-06-09 at 4 17 30 PM" src="https://github.com/chef/automate/assets/97166721/a5bda1f3-359e-4c9f-b18a-fb532b5dac7a">
<img width="1172" alt="Screenshot 2023-06-09 at 4 18 25 PM" src="https://github.com/chef/automate/assets/97166721/87e6defd-0d60-4233-8ecc-80006b6d0e0e">
